cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -174,6 +174,40 @@ func (s *Supervisor) getBootstrapInfo() (err error) {
 	return nil
 }
 
+func (s *Supervisor) Capabilities() protobufs.AgentCapabilities {
+	var supportedCapabilities protobufs.AgentCapabilities
+
+	if c := s.config.Capabilities; c != nil {
+
+		// ReportsEffectiveConfig is set if unspecified or explicitly set to true.
+		if c.ReportsEffectiveConfig != nil && *c.ReportsEffectiveConfig {
```

Comment:
```suggestion
		if (c.ReportsEffectiveConfig != nil && *c.ReportsEffectiveConfig) || c.ReportsEffectiveConfig == nil  {
```
Could we consolidate the if-statement logic here?

```golang
		if !(c.ReportsEffectiveConfig != nil && !*c.ReportsEffectiveConfig)  {
```
I think this would also work, but it isn't quite as explicit about the cases where we want to set the capability.

Reply:
Updated, I think it should be simple so that reader understands the code easily.